var/spack/repos/builtin/packages/kokkos/package.py

```diff
@@ -213,6 +213,9 @@ class Kokkos(CMakePackage, CudaPackage, ROCmPackage):
     depends_on("kokkos-nvcc-wrapper@master", when="@master+wrapper")
     conflicts("+wrapper", when="~cuda")
 
+    conflicts("+rocthrust", when="@4.3: ~rocm")
+    conflicts("~rocthrust", when="@4.3: +rocm")
```

Comment:
```suggestion
    conflicts("+rocthrust", when="@4.3: ~rocm")
```

We do not have problem compiling kokkos without `rocthrust`.